src/elements/content-sidebar/DocGenSidebar/DocGenSidebar.tsx

```diff
@@ -0,0 +1,162 @@
+        image: DocGenTag[];
+        text: DocGenTag[];
+    };
+    jsonPaths: {
+        textTree: JsonData;
+        imageTree: JsonData;
+    };
+};
+
+const DocGenSidebar = (props: Props) => {
+    const [sidebarState, setSidebarState] = React.useState<State>({
+        hasError: false,
+        loading: false,
+        tags: {
+            text: [],
+            image: [],
+        },
+        jsonPaths: {
+            textTree: {},
+            imageTree: {},
+        },
+    });
+
+    const tagsToJsonPaths = (tags: DocGenTag[]) => {
+        /* eslint-disable  @typescript-eslint/no-explicit-any */
+        const jsonPathsMap: any = {};
+        tags.forEach(tag => {
+            tag.json_paths.forEach(jsonPath => {
+                const paths = jsonPath.split('.');
+                let currentPath = '';
+                paths.forEach((segment, index) => {
+                    // avoid dot at the beginning of the path
+                    currentPath += (index > 0 ? '.' : '') + segment;
+
+                    // If the currenetPath doesn't exist in the map, initialize it
+
+                    if (!jsonPathsMap[currentPath]) {
+                        jsonPathsMap[currentPath] = [];
+                    }
+
+                    // add the tag content to the array for this path
+                    jsonPathsMap[currentPath].push(tag.tag_content);
+                });
+            });
+        });
+        return jsonPathsMap;
+    };
+
+    const loadTags = () => {
+        if (props.getDocGenTags) {
+            setSidebarState({ ...sidebarState, loading: true });
+            props
+                .getDocGenTags()
+                .then((response: DocGenTemplateTagsResponse) => {
+                    if (response) {
+                        // anything that is not an image tag for this view is treated as a text ta
+                        const textTags = response.data.filter(tag => tag.tag_type !== 'image');
+                        const imageTags = response.data.filter(tag => tag.tag_type === 'image');
+                        setSidebarState({
+                            ...sidebarState,
+                            tags: {
+                                text: textTags,
+                                image: imageTags,
+                            },
+                            loading: false,
+                            jsonPaths: {
+                                textTree: tagsToJsonPaths(textTags),
+                                imageTree: tagsToJsonPaths(imageTags),
+                            },
+                        });
+                    } else {
+                        setSidebarState({ ...sidebarState, loading: false });
+                    }
+                })
+                .catch(() => setSidebarState({ ...sidebarState, loading: false, hasError: true }));
+        }
+    };
+
+    React.useEffect(() => {
+        loadTags();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
no because we still get an infinite loop since loadTags resets the sidebarState which it depends on, which causes re-instantiation of loadTags which causes it to be called again.